modules/navier_stokes/src/fvkernels/PINSFVMomentumPressureFlux.C

```diff
@@ -57,11 +56,25 @@ ADReal
 PINSFVMomentumPressureFlux::computeQpResidual()
 {
   ADReal eps_p_interface;
+  // Momentum and porosity domains should match
+  const auto & face_type = _face_info->faceType(_var.name());
+  const bool use_elem = (face_type == FaceInfo::VarFaceNeighbors::ELEM) ||
+                        (face_type == FaceInfo::VarFaceNeighbors::BOTH);
+  const auto * const elem_ptr = use_elem ? &_face_info->elem() : _face_info->neighborPtr();
+  const auto * neighbor_ptr = use_elem ? _face_info->neighborPtr() : &_face_info->elem();
 
+  // At an external boundary, use the element where porosity is defined
+  if (!neighbor_ptr)
+    neighbor_ptr = elem_ptr;
+  // At a block restriction boundary for porosity, use porosity where it's defined
+  else if (face_type == FaceInfo::VarFaceNeighbors::ELEM)
+    neighbor_ptr = elem_ptr;
+
+  // This could be simplified if eps * P was a functor, or if we returned eps(face) * P(face)
   Moose::FV::interpolate(Moose::FV::InterpMethod::Average,
                          eps_p_interface,
-                         _eps[_qp] * _p_elem[_qp],
-                         _eps_neighbor[_qp] * _p_neighbor[_qp],
+                         _eps(makeElemArg(elem_ptr)) * _p_elem[_qp],
+                         _eps(makeElemArg(neighbor_ptr)) * _p_neighbor[_qp],
                          *_face_info,
                          true);
```

Comment:
You could create a `PiecewiseByBlockLambdaFunctor` for the aggregate here locally in the constructor. And then with the changes in https://github.com/idaholab/moose/pull/20504 you could call with the face argument and get the exact same answer as you get with this current code

Reply:
Actually you don't even need the changes in #20504 since we already support calling `PiecewiseByBlockLambdaFunctor::evaluateGradient(FaceArg)` with central difference and upwind limiting

Reply:
Not sure I understand what you mean, there's no gradient here?
Also we dont already support the faceArg for this functor? This is what I'm working on adding today?

Reply:
I made a boo-boo. I mean the value-type basic `evaluate(FaceArg)`

Reply:
ah ok. Yeah it's a pretty cool idea. Current one seems fine though, I d rather work on something else